tests/integration/logging/monolog3/test_monolog_decoration_basic.php

```diff
@@ -0,0 +1,208 @@
+    [{"name": "DurationByCaller/Unknown/Unknown/Unknown/Unknown/allOther"},       [1, "??", "??", "??", "??", "??"]],
+    [{"name": "Logging/lines"},                                                   [8, "??", "??", "??", "??", "??"]],
+    [{"name": "Logging/lines/ALERT"},                                             [1, "??", "??", "??", "??", "??"]],
+    [{"name": "Logging/lines/CRITICAL"},                                          [1, "??", "??", "??", "??", "??"]],
+    [{"name": "Logging/lines/DEBUG"},                                             [1, "??", "??", "??", "??", "??"]],
+    [{"name": "Logging/lines/EMERGENCY"},                                         [1, "??", "??", "??", "??", "??"]],
+    [{"name": "Logging/lines/ERROR"},                                             [1, "??", "??", "??", "??", "??"]],
+    [{"name": "Logging/lines/INFO"},                                              [1, "??", "??", "??", "??", "??"]],
+    [{"name": "Logging/lines/NOTICE"},                                            [1, "??", "??", "??", "??", "??"]],
+    [{"name": "Logging/lines/WARNING"},                                           [1, "??", "??", "??", "??", "??"]],
+    [{"name": "OtherTransaction/all"},                                            [1, "??", "??", "??", "??", "??"]],
+    [{"name": "OtherTransaction/php__FILE__"},                                    [1, "??", "??", "??", "??", "??"]],
+    [{"name": "OtherTransactionTotalTime"},                                       [1, "??", "??", "??", "??", "??"]],
+    [{"name": "OtherTransactionTotalTime/php__FILE__"},                           [1, "??", "??", "??", "??", "??"]],
+    [{"name": "Supportability/api/get_linking_metadata"},                         [16, "??", "??", "??", "??", "??"]],
+    [{"name": "Supportability/Logging/PHP/Monolog/enabled"},                      [1, "??", "??", "??", "??", "??"]],
+    [{"name": "Supportability/library/Monolog/detected"},                         [1, "??", "??", "??", "??", "??"]],
+    [{"name": "Supportability/Logging/LocalDecorating/PHP/enabled"},              [1, "??", "??", "??", "??", "??"]],
+    [{"name": "Supportability/Logging/Forwarding/PHP/disabled"},                  [1, "??", "??", "??", "??", "??"]],
+    [{"name": "Supportability/Logging/Metrics/PHP/enabled"},                      [1, "??", "??", "??", "??", "??"]]
+  ]
+]
+*/
+
+
+/*EXPECT_LOG_EVENTS
+null
+*/
+
+require_once(realpath(dirname(__FILE__)) . '/../../../include/config.php');
+require_once(realpath (dirname ( __FILE__ )) . '/../../../include/tap.php');
+require_once(realpath(dirname(__FILE__)) . '/../../../include/monolog.php');
+require_monolog(3);
+
+use Monolog\Logger;
+use Monolog\LogRecord;
+use Monolog\Handler\StreamHandler;
+
+/* create formatter class that echos the interesting data that the log 
+   decoration should have added. */
+class CheckDecorateFormatter implements Monolog\Formatter\FormatterInterface {
```

Comment:
Isn't this function exactly the same in all 4 tests? Could it be maybe moved to `tests/include/monolog.php` to avoid code duplication?

Reply:
I think that is a bit of a chicken and egg problem as monolog.php is primarily about provisioning Monolog and the CheckDecorateFormatter class requires Monolog to already exist.
I could move it to an include in each monolog test directory to reduce duplication.

Reply:
The Monolog2 and Monolog3 tests requires a slightly different class also.
I reduced the code duplication to some degree with this [commit](https://github.com/newrelic/newrelic-php-agent/commit/9840a6712ba2a0f4ecc12f358132cd6854e09bc0).

Reply:
Also addressed in commit [29a4f1](https://github.com/newrelic/newrelic-php-agent/commit/29a4f1bd458410cf447493328c45d7d37c46f23b).

Reply:
Thank you!